src/scarabee/_scarabee/cmfd.cpp

```diff
@@ -234,6 +290,23 @@ std::size_t CMFD::moc_to_cmfd_group(std::size_t g) const {
   return moc_to_cmfd_group_map_[g];
 }
 
+const double& CMFD::flux(const std::size_t i, const std::size_t j,
+                         const std::size_t g) const {
+  if (g > ng_) {
+    auto mssg = "Group index out of range.";
+    spdlog::error(mssg);
+    throw ScarabeeException(mssg);
+  }
+  if (i >= nx_ || j >= ny_) {
+    auto mssg = "Indexed cell does not exist.";
+    spdlog::error(mssg);
+    throw ScarabeeException(mssg);
+  }
```

Comment:
Check the i and j separately so that you can say which index is out of range to the user.